README.md

```diff
@@ -1,3 +1,142 @@
+    - dynamo TABLE_NAME (where all credentials are set via AWS' standard environment variables)
+    - badger DB_PATH
+
+### Example
+
+Below is a basic example of starting a DNS server that handles the IP based domain names as well as ACME challenges.
+It does the following:
+- Handles IP-based names and ACME challenges for the libp2p.direct forge
+- Sets up a standard HTTPS listener for registration.libp2p.direct to handle setting ACME challenges
+- Uses dynamo as a backend for ACME challenges
+
+``` corefile
+. {
+    log
+    ipparser libp2p.direct
+    acme libp2p.direct {
+        registration-domain registration.libp2p.direct listen-address=:443 external-tls=false
+        database-type dynamo mytable
+    }
+}
+```
+
+### Handled DNS records
+
+There are 3 types of records handled for a given peer and forge (e.g. `<peerID>.libp2p.direct`):
+- ACME Challenges for a given peerID `_acme-challenge.<peerID>.libp2p.direct`
+- A records for an IPv4 prefixed subdomain like `1-2-3-4.<peerID>.libp2p.direct`
+- AAAA records for an IPv6 prefixed subdomain like `2001-db8--.<peerID>.libp2p.direct`
+
+#### IPv4 subdomain handling
+
+IPv4 handling is fairly straightforward, for a given IPv4 address `1.2.3.4` convert the `.`s into `-`s and the result
+will be valid.
+
+#### IPv6 subdomain handling
+
+Due to the length of IPv6 addresses there are a number of different formats for describing IPv6 addresses.
+
+The addresses handled here are:
+- For an address `A:B:C:D:1:2:3:4` convert the `:`s into `-`s and the result will be valid.
+- Addresses of the form `A::C:D` can be converted either into their expanded form or into a condensed form by replacing
+the `:`s with `-`s, like `A--C-D`
+- When there is a `:` as the first or last character it must be converted to a 0 to comply with [rfc1123](https://datatracker.ietf.org/doc/html/rfc1123#section-2)
+, so `::B:C:D` would become `0--B-C-D` and `1::` would become `1--0`
+
+Other address formats (e.g. the dual IPv6/IPv4 format) are not supported
+
+### Submitting Challenge Records
+
+To claim a domain name like `<peerID>.libp2p.direct` requires:
+1. The private key corresponding to the given peerID
+2. A publicly reachable libp2p endpoint with 
+   - one of the following libp2p transport configurations:
+     - QUIC-v1
+     - TCP or WS or WSS, Yamux, TLS or Noise
+     - WebTransport
+     - Other transports are under consideration (e.g. HTTP), if they are of interest please file an issue
+   - the [Identify protocol](https://github.com/libp2p/specs/tree/master/identify) (`/ipfs/id/1.0.0`)
+
+To set an ACME challenge send an HTTP request to the server (for libp2p.direct this is registration.libp2p.direct)
+```shell
+curl -X POST "https://registration.libp2p.direct/v1/<peerID>/_acme-challenge" \
+-H "Authorization: libp2p-PeerID bearer=\"<base64-encoded-opaque-blob>\""
+-H "Content-Type: application/json" \
+-d '{
+  "value": "your_acme_challenge_token",
+  "addresses": "[your_multiaddrs, comma_separated]"
```

Comment:
Is `addresses` a string that contains an string array in JSON notation, eg:

```json
{
  "value": "...",
  "addresses": "[\"/ip4/...\", \"/ip6/...\"]"
}
```

or is it a single string, eg:

```json
{
  "value": "...",
  "addresses": "/ip4/...,/ip6/..."
}
```

If so, why not just a simple array of strings, eg:

```json
{
  "value": "...",
  "addresses": ["/ip4/...", "/ip6/..."]
}
```

Reply:
Looking at [the code](https://github.com/ipshipyard/p2p-forge/pull/1/files#diff-1db9aa3c1835054d2178f88673e32e5f18ed6b3a8e787df6865b89fe3e9b90eeR51) I think it is an array?

The example might be better as:

```suggestion
  "addresses": ["your", "multiaddrs"]
```